src/app/services/registry.service.ts

```diff
@@ -483,7 +483,13 @@ export class PluginRegistryBaseService {
 
     public async getByApiLink<T>(link: ApiLink, searchParams: URLSearchParams | null = null, ignoreCache: boolean | "ignore-embedded" = false): Promise<ApiResponse<T> | null> {
         const url = new URL(link.href)
-        searchParams?.forEach((value, key) => url.searchParams.append(key, value));
+        searchParams?.forEach((value, key) => {
+            if (url.searchParams.has(key)) {
+                url.searchParams.set(key, value);
+            } else {
+                url.searchParams.append(key, value);
```

Comment:
What is the difference between `set` and `append`? Maybe we can just use `set` and skip the `has` test.

Reply:
You're right, we can just use `set`.

> The [set()](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/set) method of the URLSearchParams interface sets the value associated with a given search parameter to the given value. If there were several matching values, this method deletes the others. If the search parameter doesn't exist, this method creates it.